phoenix-core/src/it/java/org/apache/phoenix/rpc/UpdateCacheIT.java

```diff
@@ -269,4 +317,49 @@ public void testInvalidConnUpdateCacheFrequencyShouldThrow() throws Exception {
             }
         }
     }
+
+    /**
+     * Helper Runnable impl class that continuously keeps checking
+     * if SYSTEM.MUTEX contains any record until either interrupted or
+     * provided connection is closed
+     */
+    private static class SystemMutexCaller implements Runnable {
```

Comment:
Perhaps a cleaner way to test this instead of a background thread that continuously queries SYSTEM.MUTEX is to use your own TestQueryServices impl (some tests do this so you can add your code there) and override the `writeMutexCell()` method to make sure if it is called. We can rely on this method since afaik we don't do checkAndPut calls to SYSTEM.MUTEX outside of this API

Reply:
Nice, that sounds good way for sure. However, do you think keeping this test as is (select queries) might be beneficial based on
1. Completeness of IT test
2. If by any chance we introduce checkAndPut on SYSTEM.MUTEX outside this API maybe with some diff value or maybe by using column value in generating rowkey, this specific test that focus on rowkey (tableName+schemaName as non-null values) might not get impacted.

Althought 2) looks highly unlikely, maybe for completeness of IT, we can keep this as is?

However, overriding `writeMutexCell()` sounds really nice one and less complex. Let me explore tomorrow and get back, but somehow I feel completeness with select query with specific where clause (as per your review comment, updated select query with where clause to focus on specific row rather than generic select query which would not work if this test is run in parallel with other tests) might be better test.

Thanks

Reply:
@ChinmaySKulkarni We can use `DelegateConnectionQueryServices` and override `writeMutexCell()`, sounds good, but I still feel from completeness of IT test viewpoint, current one (threads scanning SYSTEM.MUTEX with specific rowkey) could be better test. Thought?

Moreover, it is also confirmed that this same test fails 100% of times without source changes in this PR. Tested in multiple runs.

Reply:
I see you have modified the sys.mutex query to be more selective, so this should be fine.